modular_skyrat/code/modules/research/xenoarch/strange_rock.dm

```diff
@@ -0,0 +1,262 @@
+/obj/item/strangerock
+	icon = 'modular_skyrat/code/modules/research/xenoarch/fossil_and_artifact.dmi'
+	name = "strange rock"
+	desc = "This is a strange rock, it appears to have a relic encased."
+	icon_state = "strange"
+	item_state = "strange"
+
+	var/chosenitem = null
+	var/itemsafedepth = null
+	var/itembasedepth = null
+	var/itemactualdepth = null
+	var/dugdepth = null
+
+	var/tryagain = null
+
+/obj/item/strangerock/Initialize()
+	icon_state = pick("strange","strange0","strange1","strange2","strange3")
+	var/randomnumber = rand(1,200)
+	switch(randomnumber)
+		if(1 to 159)
+			chosenitem = pickweight(GLOB.bas_artifact)
+			itembasedepth = rand(30,40)
+			itemsafedepth = rand(3,6)
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+		if(160 to 199)
+			chosenitem = pickweight(GLOB.adv_artifact)
+			itembasedepth = rand(40,60)
+			itemsafedepth = rand(6,12)
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+		if(200)
+			chosenitem = pickweight(GLOB.ult_artifact)
+			itembasedepth = rand(70,100)
+			itemsafedepth = rand(12,14)
+			itemactualdepth = rand(itembasedepth - itemsafedepth,itembasedepth)
+	..()
+
+/obj/item/strangerock/attackby(obj/item/W, mob/user, params)
+	if(istype(W,/obj/item/xenoarch/clean/hammer/cm1))
```

Comment:
Line 38 all the way to line 157, all 120 of them, could be condensed into only 26 lines.

```suggestion
	if(istype(W, /obj/item/xenoarch/clean/hammer/))
		if(tryagain)
			to_chat(user,"You are already mining this.")
			return
		tryagain = TRUE
		var/obj/item/xenoarch/clean/hammer/HM = W
		playsound(loc, HM.usesound, 50, 1, -1)
		if(!do_after(user,HM.cleandepth * 4,target = src))
			to_chat(user,"You must stand still to clean.")
			tryagain = FALSE
			return
		dugdepth += HM.cleandepth
		if(dugdepth > itemactualdepth)
			to_chat(user,"The strange rock crumbles, destroying anything that could have been recovered.")
			qdel(src)
			return
		tryagain = FALSE
```

It's all duplicate code. You could even add 'cleanspeed' or something as a variable to the hammers themselves and use that as a modifier in the do_after if you want to keep that.

Reply:
working on this. I can condense the hammers, dont feel like condensing the brushes at the moment